packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -255,7 +329,15 @@ abstract class Route<T> {
   ///
   /// If this is true, then [isActive] is also true.
   bool get isCurrent {
-    return _navigator != null && _navigator._history.last == this;
+    if (_navigator == null)
+      return false;
+    final _RouteEntry currentRouteEntry = _navigator._history.lastWhere(
```

Comment:
Do you think this would read more clearly as

```dart
  for (_RouteEntry routeEntry in _navigator.history.reverse()) {
    if (routeEntry == this && _RouteEntry.isPresentPredicate(routeEntry) {
      return true;
    }
  }
  return false;
```

Or otherwise just combining the `== this` check with the `_RouteEntry.isPresentPredicate`?

Same for below where you iterate from front to back.

Reply:
I see them as about the same in terms of readability. I'll leave it as is.